ludwig/data/preprocessing.py

```diff
@@ -382,139 +393,31 @@ def preprocess_for_training(
         build_dataset_df
     ) = get_dataset_fun(dataset_type)
 
-    if data_df is not None:
-        # needs preprocessing
-        logging.info('Using full dataframe')
-        logging.info('Building dataset (it may take a while)')
-        data, train_set_metadata = build_dataset_df(
-            data_df,
-            features,
-            preprocessing_params,
-            random_seed=random_seed
-        )
-        if not skip_save_processed_input:
-            logging.info('Writing dataset')
-            data_utils.save_hdf5(data_hdf5_fp, data, train_set_metadata)
-        logging.info('Writing train set metadata with vocabulary')
-        data_utils.save_json(train_set_metadata_json_fp, train_set_metadata)
-        training_set, test_set, validation_set = split_dataset_tvt(
-            data,
-            data['split']
-        )
-
-    elif data_train_df is not None:
-        # needs preprocessing
-        logging.info('Using training dataframe')
-        logging.info('Building dataset (it may take a while)')
-        concatenated_df = concatenate_df(
-            data_train_df,
-            data_validation_df,
-            data_test_df
-        )
-        data, train_set_metadata = build_dataset_df(
-            concatenated_df,
-            features,
-            preprocessing_params,
-            random_seed=random_seed
-        )
-        training_set, test_set, validation_set = split_dataset_tvt(
-            data,
-            data['split']
-        )
-        if not skip_save_processed_input:
-            logging.info('Writing dataset')
-            data_utils.save_hdf5(
-                data_train_hdf5_fp,
-                training_set,
-                train_set_metadata
+    if data_df is not None or data_train_df is not None:
+        # Preprocess data frames
+        training_set, test_set, validation_set, train_set_metadata = \
```

Comment:
Nit: don't use backslashes. Can do this instead
(
    training_set,
    test_set,
    validation_set,
    train_set_metadata
) = (
    ....
)